src/components/input/Input.tsx

```diff
@@ -111,39 +120,52 @@ export class Input extends React.Component<IInputProps, any> {
             : null;
     }
 
-    render() {
+    private getRawInput(): JSX.Element {
+        const innerInputClasses = classNames({
+            invalid: !this.props.valid && contains(['number', 'text'], this.props.type),
+        }, this.props.innerInputClasses);
+
+        return (
+            <input
+                id={this.props.id}
+                className={innerInputClasses}
+                type={this.props.type}
+                defaultValue={!isUndefined(this.props.value) ? this.props.value : this.props.defaultValue}
+                ref={(innerInput: HTMLInputElement) => this.innerInput = innerInput}
+                onBlur={() => this.handleBlur()}
+                onChange={(event: React.ChangeEvent<HTMLInputElement>) => this.handleChange(event)}
+                onKeyUp={(event: React.KeyboardEvent<HTMLInputElement>) => this.handleKeyUp(event)}
+                placeholder={this.props.placeholder}
+                checked={!!this.props.checked}
+                disabled={!!this.props.disabled}
+                name={this.props.name}
+                required
+                readOnly={!!this.props.readOnly}
+            />
+        );
+    }
+
+    private getDefaultInput(): JSX.Element {
         const classes = classNames(
             'input-wrapper validate',
             {
                 'input-field': contains(['number', 'text'], this.props.type),
             },
             this.props.classes,
         );
-        const innerInputClasses = classNames({
-            invalid: !this.props.valid && contains(['number', 'text'], this.props.type),
-        }, this.props.innerInputClasses);
 
         return (
             <div className={classes} onClick={(e: React.MouseEvent<HTMLElement>) => this.handleClick(e)}>
-                <input
-                    id={this.props.id}
-                    className={innerInputClasses}
-                    type={this.props.type}
-                    defaultValue={!isUndefined(this.props.value) ? this.props.value : this.props.defaultValue}
-                    ref={(innerInput: HTMLInputElement) => this.innerInput = innerInput}
-                    onBlur={() => this.handleBlur()}
-                    onChange={() => this.handleChange()}
-                    onKeyUp={(event: React.KeyboardEvent<HTMLInputElement>) => this.handleKeyUp(event)}
-                    placeholder={this.props.placeholder}
-                    checked={!!this.props.checked}
-                    disabled={!!this.props.disabled}
-                    name={this.props.name}
-                    required
-                    readOnly={!!this.props.readOnly}
-                />
+                {this.getRawInput()}
                 {this.getLabel()}
                 {this.props.children}
             </div>
         );
     }
+
+    render() {
+        return this.props.rawInput
+            ? this.getRawInput()
+            : this.getDefaultInput();
+    }

```

Comment:
~~What's the difference between getting the RawInput and getting the DefaultInput if you don't have `children`, `labelTitle` and `labelProps` ?~~

It might be worth it to create a `RawInput` component, and use it in the Input and elsewhere where you need it. In your current solution you're ignoring a lot of props and it seems counter intuitive to me

Reply:
I currently disagree with your proposition but maybe if you give me more details on how you see the implementation i'll change my mind.  
  
I need most of the business logic already implemented by `Input` and `InputConnected`... i just need the view layer to change a little and I think it is not an anti-pattern.  Take https://github.com/buildo/react-placeholder for example, if you pass a placeholder type it will render a certain way, but if you pass a customPlaceholder it will be rendered another way, and some props won't be considered in this context too. 
  
I'll stop here and wait for your explanation to make sure I understand what you want correctly.

Reply:
Maybe it is just the naming that is confusing. I am more used to a label + Input being called a Field, while an input for me is by default "raw". But IMO the simplest implementation is often the best and what we have here is pretty simple, maybe make the naming clearer if possible. (or just a comment saying what raw means)

Reply:
I can't really rename the Input class though as I think it is used in many places outside react-vapor now, I'll add a comment for for the rawInput prop

Reply:
You could create `DefaultInput` and `RawInput` and `Input` would simply be the boolean that renders either one or the other. This way it will be easier to know the `RawInputProps`, but you'll be able to get everything you need from `InputConnected`. It will also be useful if someone ever needs the raw input being non-connected.

Reply:
added a comment for rawInput 
![screen shot 2018-03-26 at 10 01 33 am](https://user-images.githubusercontent.com/9539763/37911045-05e232fc-30dd-11e8-9e49-10cd06e1c2f5.png)
  
but will also dig the Raw/Default input thing

Reply:
I like this